ml-prims/src/array/array.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <thrust/execution_policy.h>
+#include <thrust/device_ptr.h>
+#include <thrust/unique.h>
+#include <thrust/sort.h>
+#include <limits>
+
+#include "cuda_utils.h"
+
+
+namespace MLCommon {
+namespace Array {
+
+template <typename Type, int TPB_X, typename Lambda>
+__global__ void map_label_kernel(Type *map_ids, Type *in, Type *out,
+        size_t N, Lambda filter_op) {
+    int tid = threadIdx.x + blockIdx.x*TPB_X;
```

Comment:
This is one of the reasons I used `size_t` for the `N` parameter, rather than int. I agree, it probably would be better to make `N` a template type. 

Why make `tid` configurable, though?